### DISCUSSION TITLE
NextCloud on PI 4 with 8GB (arm/v7) does not work

### DISCUSSION THREAD

**Discussion by msa7** [nextcloud/all-in-one#1405] [Questions] [answered]
Hi. I am trying to install and run NextCloud on my Raspberry PI 4 with 8GB of ram. I got an error `exec format error`:

```
pi@xranitel:~/prj/next_cloud $     docker run \
>     --sig-proxy=false \
>     --name nextcloud-aio-mastercontainer \
>     --restart always \
>     --publish 8080:8080 \
>     -e APACHE_PORT=11000 \
>     --volume nextcloud_aio_mastercontainer:/mnt/xranitel/nextcloud \
>     --volume /var/run/docker.sock:/var/run/docker.sock:ro \
>     nextcloud/all-in-one:latest-arm64
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/arm/v7) and no specific platform was requested
exec /usr/bin/start.sh: exec format error
pi@xranitel:~/prj/next_cloud $
```

Any idea how to process?

Thanks!

**Comment by szaimen** [comment, marked as answer]
> Any idea how to process?

Yes, your OS is most likely not 64-bit.

**Reply by msa7** [reply]
yep, thanks. I reinstall PI with 64bit, it works now

**Reply by metawops** [reply]
So does this mean that Nextcloud can't be installed on an armv7l architecture at all? Or is there a docker image of Nextcloud AIO for this type of system?

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one#which-cpu-architectures-are-supported

**Reply by metawops** [reply]
Oh my. How disappointing. I was hoping to be able to replace my !"§$%& ownCloud installation that actually runs on this Raspberry Pi 4 armv7l with a nice new Nextcloud installation. 😞

**Reply by szaimen** [reply]
A raspberry Pi 4 actually is aarch64 and arm64, only needs a 64-bit OS.

**Reply by metawops** [reply]
Unfortunately, an upgrade of the OS (Raspberry Pi OS) from 32 to 64 bit doesn't seem to be easy. There's other software running on this Raspi4 that needs to still run afterwards ... 😳

**Reply by Zoey2936** [reply]
sometime ago I switched from 32bit to 64bit on my pi4 by just doing `echo "arm_64bit=1" | tee -a /boot/config.txt && reboot`

**Reply by metawops** [reply]
That will only change the kernel to 64 bit (if /boot/kernel8.img exists). However, most apps won't work anymore because no 64 bit libraries will be there and have to be installed.
